extensions/amp-stream-gallery/1.0/stream-gallery.js

```diff
@@ -91,12 +92,23 @@ export function StreamGallery(props) {
     ]
   );
 
+  useImperativeHandle(
+    ref,
+    () =>
+      /** @type {!BaseCarouselDef.CarouselApi} */ ({
+        goToSlide: carouselRef.current.goToSlide,
```

Comment:
One thing to note. The `goToSlide: carouselRef.current.goToSlide` is not the same as `goToSlide: (...) => carouselRef.current.goToSlide(...)`. The main difference is that the second option "observes" the changes to the `carouselRef.current` instance, while the first form only picks some the first instance and uses it regardless whether `carouselRef.current` ever changes. For this reason, the second form would be safer, since we can't always rely on the "stability" of a `ref` value.

Reply:
Is it because the second form always freshly references the `carouselRef.current` in the call, whereas the first form stores a copy of the fn as-is at time of first reference for future use?

Reply:
Yep. Exactly.